services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/application/ApplicationServiceITest.kt

```diff
@@ -926,6 +929,44 @@ class ApplicationServiceITest : DatabaseTest() {
         }
     }
 
+    @Test
+    @Sql("/sql/senaatintorin-hanke.sql")
+    fun `sendApplication saves user tokens from application contacts`() {
+        val application =
+            alluDataFactory.saveApplicationEntity(
+                USERNAME,
+                hanke = initializedHanke(),
+                application = mockApplicationWithArea()
+            ) { it.alluid = null }
+        val applicationData = application.applicationData as CableReportApplicationData
+        val pendingApplicationData = applicationData.copy(pendingOnClient = false)
+        every { cableReportServiceAllu.create(pendingApplicationData.toAlluData()) } returns 26
+        justRun { cableReportServiceAllu.addAttachment(26, any()) }
+        every { cableReportServiceAllu.getApplicationInformation(26) } returns
+            AlluDataFactory.createAlluApplicationResponse(26)
+
+        applicationService.sendApplication(application.id!!, USERNAME)
+
+        val tunnisteet = kayttajaTunnisteRepository.findAll()
+        assertThat(tunnisteet).hasSize(1)
+        assertThat(tunnisteet[0].role).isEqualTo(Role.KATSELUOIKEUS)
+        assertThat(tunnisteet[0].createdAt).isRecent()
+        assertThat(tunnisteet[0].sentAt).isNull()
+        assertThat(tunnisteet[0].tunniste).matches(Regex("[a-zA-z0-9]{24}"))
```

Comment:
The regex could be defined as a variable with a describing name (If possible).

Reply:
Moved to a variable.